streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -177,10 +199,29 @@ void openDB(final ProcessorContext context) {
             throw new ProcessorStateException(fatal);
         }
 
+        metricsRecorder.ifPresent(
+            metricsRecorder -> setUpMetrics(
+                (StreamsMetricsImpl) context.metrics(),
+                context.taskId(),
+                RecordingLevel.forName((String) configs.get(METRICS_RECORDING_LEVEL_CONFIG))
+            )
+        );
+
         openRocksDB(dbOptions, columnFamilyOptions);
         open = true;
     }
 
+    private void setUpMetrics(final StreamsMetricsImpl metrics,
+                              final TaskId taskId,
+                              final RecordingLevel recordingLevel) {
+        if (userSpecifiedOptions.statistics() == null && recordingLevel == RecordingLevel.DEBUG) {
```

Comment:
should we record metrics even when the userSpecifiedOptions already has it configured? Or is it better not to, because we don't know what kind of configuration they gave it?